### PR TITLE
feat: generate custom asset logo when not available

### LINKED ISSUE #3999: [Task]: Use initials for token icon if logo is not available
### Task description

We need should use the initials or the symbol to create an icon for tokens where the logo is not available or not displayable.
We will limit this to 5 characters.

### Requirements

- [ ] If logo not available add initials or symbol as the icon
- [ ] Change text size depending on length
- [ ] Give a token a random colour when we persist the metadata

### Creation checklist

- [ ] I have assigned this task to the correct people
- [ ] I have added the most appropriate labels
- [ ] I have linked the correct milestone and/or project

### PR DESCRIPTION
**Pull request by begonaalvarezd**
## Summary

> Please summarize your changes, describing __what__ they are and __why__ they were made.

## Changelog

```
- Please write a list of changes
```

## Relevant Issues

> Please list any related issues using [development keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

Close #3999

### Platforms

> Please select any platforms where your changes have been tested.

- __Desktop__
  - [ ] MacOS
  - [x] Linux
  - [ ] Windows
- __Mobile__
  - [ ] iOS
  - [ ] Android

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to __test and verify__ that your changes work as intended.

...

## Checklist

> Please tick all of the following boxes that are relevant to your changes.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or modified tests that prove my changes work as intended
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have verified that my latest changes pass CI workflows for testing and linting
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/shared/components/molecules/AssetIcon.svelte [hunk 1]
```diff
@@ -1,24 +1,47 @@
...
+    import { updatePersistedAsset } from '@core/wallet/stores'
+    import { getInitials, isBright } from '@lib/helpers'
     import { Icon, VerificationBadge } from 'shared/components'
+    import { onMount } from 'svelte'
 
     export let asset: IAsset
     export let large = false
     export let showVerificationBadge = false
 
+    const MAX_DISPLAYED_INITIALS = 5
```
Comment (nicole-obrien) [inline packages/shared/components/molecules/AssetIcon.svelte:14]: Lets move this to a constant file

File: packages/shared/components/molecules/AssetIcon.svelte [hunk 2]
```diff
@@ -1,24 +1,47 @@
...
-        default:
-            icon = 'tokens'
-    }
+        } else {
+            // Generate our own icon
+            // Icon content: unit or initials
+            // Icon color: deterministic color based on asset name in color range 500-800
+            assetInitials =
+                asset?.metadata?.unit?.slice(0, MAX_DISPLAYED_INITIALS) ??
+                getInitials(asset?.metadata?.name, MAX_DISPLAYED_INITIALS)
```
Comment (nicole-obrien) [inline packages/shared/components/molecules/AssetIcon.svelte:35]: Can we abstract this? Might be useful somewhere else in the future

File: packages/shared/components/molecules/AssetIcon.svelte [hunk 3]
```diff
@@ -1,24 +1,47 @@
...
+            // If the primaryColor is not yet set, find it and persist it
+            if (!asset?.metadata?.primaryColor) {
+                const colorFromString = getIconColorFromString(asset?.metadata?.name)
+                updatePersistedAsset({ id: asset?.id, metadata: { ...asset?.metadata, primaryColor: colorFromString } })
+                assetIconBackgroundColor = colorFromString
```
Comment (nicole-obrien) [inline packages/shared/components/molecules/AssetIcon.svelte:40]: Lets move this to packages/shared/lib/core/wallet/helpers/buildPersistedAssetFromIrc30Metadata.ts

**Review by nicole-obrien — CHANGES_REQUESTED** [review]
Just some refactoring to do 🙏

File: packages/shared/components/molecules/AssetIcon.svelte (hunk 1 above)
Comment (begonaalvarezd) [reply packages/shared/components/molecules/AssetIcon.svelte]: updated

File: packages/shared/components/molecules/AssetIcon.svelte (hunk 2 above)
Comment (begonaalvarezd) [reply packages/shared/components/molecules/AssetIcon.svelte]: updated

File: packages/shared/components/molecules/AssetIcon.svelte (hunk 3 above)
Comment (begonaalvarezd) [reply packages/shared/components/molecules/AssetIcon.svelte]: updated

**Review by nicole-obrien — APPROVED** [review]
Looks good to me 🔥

**Pull request merged**